x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/phase_timings_context.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { createContext, FunctionComponent, useContext } from 'react';
+import { useFormData } from '../../../../shared_imports';
+import { FormInternal } from '../types';
+import { UseField } from './index';
+
+export type PhaseTimingConfiguration = 'forever' | 'last' | 'disabled' | 'enabled';
```

Comment:
I think this approach may be tying too much overlapping state into one value. For instance; if a phase is `forever` that implies that it is also `last` and `enabled` - which does not really sit well with the individual union of types.

I propose we extract the `enabled | disabled` state from the `last | forever`. I iterated on this a little bit and came up with the following: 

<details>
<summary>diff of proposed changes</summary>

```diff
diff --git i/x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/phase_footer/phase_footer.tsx w/x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/phase_footer/phase_footer.tsx
index 880a354b4b0..0bc97e8f250 100644
--- i/x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/phase_footer/phase_footer.tsx
+++ w/x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/phase_footer/phase_footer.tsx
@@ -23,11 +23,13 @@ interface Props {
 }
 
 export const PhaseFooter: FunctionComponent<Props> = ({ phase }) => {
-  const phaseTimings = usePhaseTimings();
-  const phaseConfiguration = phaseTimings[phase];
-  const setValue = phaseTimings.setDeletePhaseEnabled;
+  const {
+    isDeletePhaseEnabled,
+    setDeletePhaseEnabled,
+    [phase]: phaseConfiguration,
+  } = usePhaseTimings();
 
-  if (phaseConfiguration === 'disabled' || phaseConfiguration === 'enabled') {
+  if (!phaseConfiguration.isFinalDataPhase) {
     return (
       <EuiPanel className={'ilmPhaseFooter'} hasShadow={false}>
         <EuiText size={'s'}>&nbsp;</EuiText>
@@ -35,7 +37,7 @@ export const PhaseFooter: FunctionComponent<Props> = ({ phase }) => {
     );
   }
 
-  if (phaseConfiguration === 'forever') {
+  if (!isDeletePhaseEnabled) {
     return (
       <EuiPanel color={'subdued'} className={'ilmPhaseFooter'} hasShadow={false}>
         <InfinityIcon size={'s'} />{' '}
@@ -44,7 +46,10 @@ export const PhaseFooter: FunctionComponent<Props> = ({ phase }) => {
             id="xpack.indexLifecycleMgmt.editPolicy.phaseTiming.foreverTimingDescription"
             defaultMessage="Data will remain in this phase forever."
           />{' '}
-          <EuiLink onClick={() => setValue(true)} data-test-subj={'enableDeletePhaseLink'}>
+          <EuiLink
+            onClick={() => setDeletePhaseEnabled(true)}
+            data-test-subj={'enableDeletePhaseLink'}
+          >
             <FormattedMessage
               id="xpack.indexLifecycleMgmt.editPolicy.deletePhase.enablePhaseButtonLabel"
               defaultMessage="Set delete age"
@@ -63,7 +68,10 @@ export const PhaseFooter: FunctionComponent<Props> = ({ phase }) => {
           id="xpack.indexLifecycleMgmt.editPolicy.phaseTiming.beforeDeleteDescription"
           defaultMessage="Data will be deleted after this phase."
         />{' '}
-        <EuiLink onClick={() => setValue(false)} data-test-subj={'disableDeletePhaseLink'}>
+        <EuiLink
+          onClick={() => setDeletePhaseEnabled(false)}
+          data-test-subj={'disableDeletePhaseLink'}
+        >
           <FormattedMessage
             id="xpack.indexLifecycleMgmt.editPolicy.deletePhase.disablePhaseButtonLabel"
             defaultMessage="Keep data"
diff --git i/x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/phase_timings_context.tsx w/x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/phase_timings_context.tsx
index 796cab1b404..fdfaec35e5b 100644
--- i/x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/phase_timings_context.tsx
+++ w/x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/phase_timings_context.tsx
@@ -5,12 +5,19 @@
  * 2.0.
  */
 
-import React, { createContext, FunctionComponent, useContext } from 'react';
+import React, { createContext, FunctionComponent, useContext, useCallback } from 'react';
 import { useFormData } from '../../../../shared_imports';
 import { FormInternal } from '../types';
 import { UseField } from './index';
 
-export type PhaseTimingConfiguration = 'forever' | 'last' | 'disabled' | 'enabled';
+// export type PhaseTimingConfiguration = 'forever' | 'last' | 'disabled' | 'enabled';
+
+export interface PhaseTimingConfiguration {
+  /**
+   * Whether this is the final, non-delete, phase.
+   */
+  isFinalDataPhase: boolean;
+}
 
 const getPhaseTimingConfiguration = (
   formData: FormInternal
@@ -21,23 +28,23 @@ const getPhaseTimingConfiguration = (
 } => {
   const isWarmPhaseEnabled = formData?._meta?.warm?.enabled;
   const isColdPhaseEnabled = formData?._meta?.cold?.enabled;
-  if (formData?._meta?.delete?.enabled) {
-    return {
-      hot: !isWarmPhaseEnabled && !isColdPhaseEnabled ? 'last' : 'enabled',
-      warm: isWarmPhaseEnabled ? (isColdPhaseEnabled ? 'enabled' : 'last') : 'disabled',
-      cold: isColdPhaseEnabled ? 'last' : 'disabled',
-    };
-  }
   return {
-    hot: !isWarmPhaseEnabled && !isColdPhaseEnabled ? 'forever' : 'enabled',
-    warm: isWarmPhaseEnabled ? (isColdPhaseEnabled ? 'enabled' : 'forever') : 'disabled',
-    cold: isColdPhaseEnabled ? 'forever' : 'disabled',
+    hot: {
+      isFinalDataPhase: !isWarmPhaseEnabled && !isColdPhaseEnabled,
+    },
+    warm: {
+      isFinalDataPhase: isWarmPhaseEnabled && !isColdPhaseEnabled,
+    },
+    cold: {
+      isFinalDataPhase: isColdPhaseEnabled,
+    },
   };
 };
 export interface PhaseTimings {
   hot: PhaseTimingConfiguration;
   warm: PhaseTimingConfiguration;
   cold: PhaseTimingConfiguration;
+  isDeletePhaseEnabled: boolean;
   setDeletePhaseEnabled: (enabled: boolean) => void;
 }
 
@@ -49,12 +56,13 @@ export const PhaseTimingsProvider: FunctionComponent = ({ children }) => {
   });
 
   return (
-    <UseField path="_meta.delete.enabled">
+    <UseField<boolean> path="_meta.delete.enabled">
       {(field) => {
         return (
           <PhaseTimingsContext.Provider
             value={{
               ...getPhaseTimingConfiguration(formData),
+              isDeletePhaseEnabled: !!field.value,
               setDeletePhaseEnabled: field.setValue,
             }}
           >

```

</details>

Let me know what you think!